netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -234,11 +243,18 @@ private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream
   /**
    * Handler for an inbound HTTP/2 DATA frame.
    */
-  private void onDataRead(int streamId, ByteBuf data, boolean endOfStream) {
+
+  private void onDataRead(int streamId, ByteBuf data, int padding, boolean endOfStream) {
+    if (!flowControlPing.isPinging()) {
```

Comment:
Could this block be moved into `flowControlPinger` and done automatically? It doesn't seem like you need the `setPinging()` randomly scattered about. If you renamed `incrementDataSincePing` to `onDataRead` (or any other similar name that is less specific about what will be done) then this block would naturally be able to be done in that method.